### PR TITLE
Add rps Project

### PR DESCRIPTION
**Pull request by SpanishTorture**

### CODE REVIEW

File: RPS Project/Game.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#Returns game result
+def judge(sign_1, sign_2):
+    result = sign_1 - sign_2
+    if abs(result) > 1:
+        result = int(-result/2)
+    return result
+
+#Strategy. Based on opponent's history, probability of each sign scales with the frequency of the sign it defeats. Chooses rock as first move.
+def stratagy():
+    if count[0]==1:
```
Comment (Nigelmnz) [inline RPS Project/Game.py]: Spaces around `==`

**Review by Nigelmnz — CHANGES_REQUESTED** [review]

File: RPS Project/Game.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    if sign == 3:
+        sys.exit("Have a nice day")
+        
+    #stores user sign
+    history[sign] += 1
+
+    winner = judge(sign,computer_sign)
+
+    #prints result
+    if winner==0:
```
Comment (Nigelmnz) [inline RPS Project/Game.py:48]: spaces around `==`

**Review by Nigelmnz — CHANGES_REQUESTED** [review]

File: RPS Project/Game.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+#Returns game result
+def judge(sign_1, sign_2):
+    result = sign_1 - sign_2
+    if abs(result) > 1:
+        result = int(-result/2)
+    return result
+
+#Strategy. Based on opponent's history, probability of each sign scales with the frequency of the sign it defeats. Chooses rock as first move.
+def stratagy():
```
Comment (Nigelmnz) [inline RPS Project/Game.py:20]: `stratagy` -> `strategy`

File: RPS Project/Game.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+score = [0,0,0]
+history = [0,0,0]
+count = [0]
+
+#Friendly greeting
+input("Good Lobning. Press enter.")
+print("Choices: \n 0 = Rock \n 1 = Paper \n 2 = Scissors \n 3= exit")
+
+#Returns game result
+def judge(sign_1, sign_2):
```
Comment (Nigelmnz) [inline RPS Project/Game.py:13]: Instead of using integers to represent R/P/S, use enums. https://docs.python.org/3/library/enum.html

This will have the advantage of being much more readable. Your strategy function would return values like `Move.ROCK` instead of `0`, for example. This function would gain readability as well, as its current functionality is a bit cryptic. You also gain some readability when accessing the history, being able to access moves as `history[Move.ROCK]`.

File: RPS Project/Game.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+score = [0,0,0]
+history = [0,0,0]
+count = [0]
+
+#Friendly greeting
+input("Good Lobning. Press enter.")
+print("Choices: \n 0 = Rock \n 1 = Paper \n 2 = Scissors \n 3= exit")
+
+#Returns game result
```
Comment (Nigelmnz) [inline RPS Project/Game.py:12]: If you are going to ever have functions with non-obvious return values, you must document them. Without looking at your `game()` function or parsing the logic of this function, there's no easy way for a developer to know that this function will return 0/1/2, and that those mean draw/win/lose.

File: RPS Project/Game.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import sys
+import random
+
+score = [0,0,0]
+history = [0,0,0]
+count = [0]
```
Comment (Nigelmnz) [inline RPS Project/Game.py:6]: Is there a point to this being an array? Also, use a more descriptive name. `game_count` is probably enough.

File: RPS Project/Game.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+#Returns game result
+def judge(sign_1, sign_2):
+    result = sign_1 - sign_2
+    if abs(result) > 1:
+        result = int(-result/2)
+    return result
+
+#Strategy. Based on opponent's history, probability of each sign scales with the frequency of the sign it defeats. Chooses rock as first move.
+def stratagy():
+    if count[0] == 1:
```
Comment (Nigelmnz) [inline RPS Project/Game.py:21]: Any reason why this shouldn't just start with a random move?

File: RPS Project/Tests.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
+import random
+
+#sets paramaters, including the probability that of the "player's" choices, and starts the game
+def test(number, prob_0, prob_1):
+    global score, history, count, rock, paper
+    score = [0,0,0]
+    history = [0,0,0]
+    count = [0]
```
Comment (Nigelmnz) [inline RPS Project/Tests.py:8]: Same thing, why is count a list?

File: RPS Project/Tests.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
+import random
+
```
Comment (Nigelmnz) [inline RPS Project/Tests.py:2]: Please use the Python unittesting framework. https://docs.python.org/3/library/unittest.html

If this is too difficult to digest, please try this: http://pythontesting.net/framework/unittest/unittest-introduction/

File: RPS Project/Tests.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    global score, history, count, rock, paper
+    score = [0,0,0]
+    history = [0,0,0]
+    count = [0]
+    rock = prob_0
+    paper = prob_1
+    game(number)
+
+#Returns game result
+def judge(sign_1, sign_2):
```
Comment (Nigelmnz) [inline RPS Project/Tests.py:14]: You should be using your tests to test `game.py`, not a copy of those functions. Import the functions you want to test.

File: RPS Project/Tests.py [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+test(1000, 0, 1)
+
+#An opponent that plays with a randomly selected bias. 
+for a in range(4):
+    rock = random.uniform(0,1)
+    paper = random.uniform(0,1-rock)
+    test(1000,rock,paper)
+
+#a random opponent
+print("Failure here is fine")
```
Comment (Nigelmnz) [inline RPS Project/Tests.py:72]: A test failure is never fine. If you know something is supposed to behave a certain way, then change the pass criteria to match that behavior. This this case, your pass condition should be that against a random opponent, your algorithm performs at around a 33% success rate. Test for that.

You probably did this wanting to reuse your test `game()`. When rewriting your test framework using unittest, a better approach will be to have a function that returns your bot's winrate against an opponent with some given parameters. Then you can easily do a test on that number against whatever you expect it to be.

File: RPS Project/Game.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    elif winner == 1:
+        print("You win")
+    else:
+        print("You lose. You suck.")
+    
+    #stores and prints cumulative score
+    score[winner] += 1
+    print("Games: ", count[0])
+    print("Result: \n draw, win, loss = ", score)
+    game()
```
Comment (Nigelmnz) [inline RPS Project/Game.py:59]: Instead of a infinitely recursive call, use a `while True` loop. If you remember the memory implications of function calls, you might be able to guess why this might be a problem.

**Review by Nigelmnz — CHANGES_REQUESTED** [review]

**Comment by Nigelmnz** [comment]
@SpanishTorture Ping
